src/concrete_solve.jl

```diff
@@ -334,6 +334,211 @@ function DiffEqBase._concrete_solve_adjoint(prob, alg,
+                                                   for i in 1:length(_out)], ts)
+        end
+        only_end = length(ts) == 1 && ts[1] == _prob.tspan[2]
+    end
+
+    _save_idxs = save_idxs === nothing ? Colon() : save_idxs
+
+    function adjoint_sensitivity_backpass(Δ)
+        function df(_out, u, p, t, i)
+            outtype = typeof(_out) <: SubArray ?
+                      DiffEqBase.parameterless_type(_out.parent) :
+                      DiffEqBase.parameterless_type(_out)
+            if only_end
+                eltype(Δ) <: NoTangent && return
+                if typeof(Δ) <: AbstractArray{<:AbstractArray} && length(Δ) == 1 && i == 1
+                    # user did sol[end] on only_end
+                    if typeof(_save_idxs) <: Number
+                        x = vec(Δ[1])
+                        _out[_save_idxs] .= adapt(outtype, @view(x[_save_idxs]))
+                    elseif _save_idxs isa Colon
+                        _out = adapt(outtype, vec(Δ[1]))
+                    else
+                        vec(@view(_out[_save_idxs])) .= adapt(outtype,
+                                                              vec(Δ[1])[_save_idxs])
+                    end
+                else
+                    Δ isa NoTangent && return
+                    if typeof(_save_idxs) <: Number
+                        x = vec(Δ)
+                        _out[_save_idxs] .= adapt(outtype, @view(x[_save_idxs]))
+                    elseif _save_idxs isa Colon
+                        _out = adapt(outtype, vec(Δ))
+                    else
+                        x = vec(Δ)
+                        vec(@view(_out[_save_idxs])) .= adapt(outtype, @view(x[_save_idxs]))
+                    end
+                end
+            else
+                !Base.isconcretetype(eltype(Δ)) &&
+                    (Δ[i] isa NoTangent || eltype(Δ) <: NoTangent) && return
+                if typeof(Δ) <: AbstractArray{<:AbstractArray} || typeof(Δ) <: DESolution
+                    x = Δ[i]
+                    if typeof(_save_idxs) <: Number
+                        _out[_save_idxs] = @view(x[_save_idxs])
+                    elseif _save_idxs isa Colon
+                        _out = vec(x)
+                    else
+                        vec(@view(_out[_save_idxs])) .= vec(@view(x[_save_idxs]))
+                    end
+                else
+                    if typeof(_save_idxs) <: Number
+                        _out[_save_idxs] = adapt(outtype,
+                                                 reshape(Δ, prod(size(Δ)[1:(end - 1)]),
+                                                         size(Δ)[end])[_save_idxs, i])
+                    elseif _save_idxs isa Colon
+                        _out = vec(adapt(outtype,
+                                         reshape(Δ, prod(size(Δ)[1:(end - 1)]),
+                                         size(Δ)[end])[:, i]))######Required Assignment#################
+                    else
```

Comment:
This is the only assignment that is required for the neural ode with SArrays to work. The others are indexing on `_out` , so, won't be compatibe with SArrays anyway. Let me know if I should simplify it to just contain this assignment.

Reply:
No it's not, it's dependent on the arguments. Did you try the other arguments?